### ISSUE TITLE
Suggestion: Make Core Functionality Into Module

### ISSUE THREAD

**Issue by nathangoenner** [ngoenner84/un_thread_calculator#2]
Just a suggestion, but you mentioned wanting to make a GUI. I think a good first step would be to wrap the core functionality (actually taking a bunch of parameters and calculating an output) into a python module, which can be imported and extended by someone using python, or by a command line wrapper, or by a GUI.

Something like a simple module that has a few functions defined that accept necessary parameters. Then the command line version (like you already implemented) can import the module, prompt the user for input, and make a call to the core module. Or, if you're running in python you could just directly call the function (skip all the tedious prompts). Or have the values generated from the GUI.

**Comment by ngoenner84** [comment]
Had to read this a number of times and look some stuff up, but I think I get the gist. If I understand correctly, creating a module, someone could import the functionality into their code similar to other modules like pandas, PyPDF2, etc. Then it's easy to use a GUI/web app/etc. to call the module and get the output. On the right track?

**Comment by nathangoenner** [comment]
Exactly. You have one file, maybe called 'unthread.py' that contains the function definitions for calculation. Then in another file, say your cmd line implementation would do something like
```python
import unthread as un

un.calc_pd_tol(md_b, LE, p)
```

It would make the project more modular and extensible, in my opinion also easier to read rather than putting everything in the main.
